R/cloud.R

```diff
@@ -527,18 +527,34 @@ cloud_email <- function(type = c("broken", "failed"), job_id = cloud_job(), pkg
 
 #' Return the current cloud job
 #'
-#' This is automatically set by [cloud_check()] and only lasts for the current R session.
+#' The `job_id` is automatically set by [cloud_check()] and is remembered for
+#' the duration of the current R session. If there is no active `job_id`, but
+#' there are local cloud check results, `job_id` is inferred from the most
+#' recently modified cloud check results.
+#'
 #' @param job_id If not `NULL`, sets the active `job_id` to the input.
+#' @inheritParams cloud_report
 #' @export
-cloud_job <- function(job_id = NULL) {
-  if (!is.null(job_id)) {
-    cloud_data$job_id <- job_id
+cloud_job <- function(job_id = NULL, pkg = ".") {
```

Comment:
Here's one infelicity about this argument order and adding an explicit `pkg = pkg` to all these default calls: it suggests that we'll actually honour `pkg` but, if we've already cached an active `job_id`, we never hit the code that consults `pkg`. But I did it anyway.